apps/web/src/pages/templates/hooks/useTemplatesStoreModal.tsx

```diff
@@ -1,23 +1,34 @@
 import { useDisclosure } from '@mantine/hooks';
 
 import { IBlueprintsGrouped } from '../../../api/hooks';
-import { useInlineComponent } from '../../../hooks';
+import { useInlineComponent, useIsTemplateStoreEnabled } from '../../../hooks';
 import { ITemplatesStoreModalProps, TemplatesStoreModal } from '../components/templates-store';
 
 const NULL_COMPONENT = () => null;
 
-export const useTemplatesStoreModal = ({ general = [] }: { general?: IBlueprintsGrouped[] }) => {
-  const [isOpened, { open, close }] = useDisclosure(false);
+export const useTemplatesStoreModal = ({
+  general = [],
+  popular,
+}: {
+  general?: IBlueprintsGrouped[];
+  popular?: IBlueprintsGrouped;
+}) => {
+  const [opened, { open, close }] = useDisclosure(false);
+  const isTemplateStoreEnabled = useIsTemplateStoreEnabled();
   const hasGroups = general && general.length > 0;
+  const hasPopular = !!(popular && popular?.blueprints.length > 0);
+  const hasGroupsOrPopular = hasGroups || hasPopular;
+  const isOpened = opened && hasGroupsOrPopular && isTemplateStoreEnabled;
```

Comment:
Looks like all this information could be living in its own hook for tidiness, though I acknowledge the reusability might be non existent.